main/audiohook.c

```diff
@@ -281,13 +281,13 @@ static struct ast_frame *audiohook_read_frame_both(struct ast_audiohook *audioho
 	}
 
 	/* If we want to provide only a read factory make sure we aren't waiting for other audio */
```

Comment:
This logic has grown complicated enough that I think comments should be added to each explaining the logic and why it is present and how it works